magnolia-vpro-ui/src/main/java/nl/vpro/magnolia/ui/irma/ProofOfProvenanceField.java

```diff
@@ -89,23 +93,17 @@ protected Component initContent() {
         Button button = new Button();
         button.setCaption("Sign with Irma");
 
-        button.addClickListener(new Button.ClickListener() {
-            @Override
-            public void buttonClick(Button.ClickEvent event) {
-                Document document = Jsoup.parse(text.getValue());
-                String attribute = definition.getDefinition().getAttribute();
-                String url = proofOfProvenanceService.getBaseUrl();
-                Page.getCurrent().getJavaScript().execute("sign('" + escapeJavaScript(url) + "','" + escapeJavaScript(document.text()) + "','" + attribute + "','" + signature.getId() + "')");
-            }
+        button.addClickListener((Button.ClickListener) event -> {
+            Document document = Jsoup.parse(text.getValue());
+            String attribute = definition.getDefinition().getAttribute();
+            String url = proofOfProvenanceService.getBaseUrl();
+            Page.getCurrent().getJavaScript().execute("sign('" + escapeJavaScript(url) + "','" + escapeJavaScript(document.text()) + "','" + attribute + "','" + signature.getId() + "')");
         });
 
         com.vaadin.ui.JavaScript.getCurrent().addFunction("nl.vpro.magnolia.ui.irma.callBack",
```

Comment:
JavaScript inserten via Java? :thinking:

Reply:
Niks mis mee in deze context, maar meer verbaasd over het bestaan ervan

Reply:
Tja, ik heb het ook niet bedacht, maar Michiel.